core/lib/multivm/src/versions/vm_latest/old_vm/event_sink.rs

```diff
@@ -63,6 +67,101 @@ impl<H: HistoryMode> InMemoryEventSink<H> {
         Self::events_and_l1_messages_from_history(self.log_queries_after_timestamp(from_timestamp))
     }
 
+    fn events_logs_from_history(history: &[Box<LogQuery>]) -> Vec<LogQuery> {
+        // Filter out all the L2->L1 logs and leave only events
+        let mut events = history
+            .iter()
+            .filter_map(|log_query| (log_query.aux_byte == EVENT_AUX_BYTE).then_some(**log_query))
+            .collect_vec();
+
+        // Sort the events by timestamp and rollback flag, basically ensuring that
+        // if an event has been rolled back, the original event and its rollback will be put together
+        events.sort_by(|a, b| match a.timestamp.0.cmp(&b.timestamp.0) {
+            Ordering::Equal => {
+                if b.rollback {
+                    Ordering::Less
+                } else {
+                    Ordering::Greater
+                }
+            }
+            r => r,
+        });
```

Comment:
more like a nit: if there are two events with the same `rollback` field, then the comparator won't be transitive https://doc.rust-lang.org/std/vec/struct.Vec.html#method.sort_by. I'd prefer us not to rely on the fact it can't happen and just sort by `(a.timestamp, a.rollback)`.